src/Microsoft.AspNetCore.Routing.Abstractions/RouteValueDictionary.cs

```diff
@@ -524,30 +524,44 @@ public void Dispose()
             {
             }
 
+            // Similar to the design of List<T>.Enumerator - Split into fast path and slow path for inlining friendliness
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public bool MoveNext()
             {
-                if (++_index < _dictionary.Count)
-                {
-                    if (_dictionary._propertyStorage != null)
-                    {
-                        var storage = _dictionary._propertyStorage;
-                        var property = storage.Properties[_index];
-                        Current = new KeyValuePair<string, object>(property.Name, property.GetValue(storage.Value));
-                        return true;
-                    }
+                var dictionary = _dictionary;
 
+                // The uncommon case is that the propertyStorage is in use
+                if (dictionary._propertyStorage == null && ((uint)_index < (uint)dictionary._count))
+                {
                     Current = _dictionary._arrayStorage[_index];
```

Comment:
Should this use the local variable? `dictionary._arrayStorage[_index];`

If there is a reason not to then explain in a comment

Reply:
ooh spicy, yeah this is a mistake.